metals/src/main/scala/scala/meta/internal/builds/MillBuildTool.scala

```diff
@@ -144,6 +145,12 @@ case class MillBuildTool(userConfig: () => UserConfiguration)
   }
 
   override val buildServerName: Option[String] = Some(MillBuildTool.name)
+
+  override def isBspCompatible(bsp: String): Boolean =
+    bsp match {
+      case SbtBuildTool.name => false
+      case _ => !ScalaCli.names.contains(bsp)
```

Comment:
```suggestion
      case _ if ScalaCli.names.contains(bsp) => false
      case _ => true
```
It's a bit more readable for me